### PR TITLE
[kube-prometheus-stack] Fix PSP deprecation after k8s 1.25+

### PR DESCRIPTION
**Pull request by QuentinBisson**
Signed-off-by: QuentinBisson <quentin@giantswarm.io>

#### What this PR does / why we need it

This PR fixes PSP deprecation after k8s 1.25+ added for the prometheus operator patch jobs.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by QuentinBisson** [comment]
@monotek do you mind reviewing this?

File: charts/kube-prometheus-stack/templates/prometheus-operator/admission-webhooks/job-patch/clusterrole.yaml [hunk 1]
```diff
@@ -18,6 +18,7 @@ rules:
     verbs:
       - get
       - update
+{{- if .Capabilities.APIVersions.Has "policy/v1beta1/PodSecurityPolicy" }}
```
Comment (monotek) [inline charts/kube-prometheus-stack/templates/prometheus-operator/admission-webhooks/job-patch/clusterrole.yaml:21]: Please use "if and"

File: charts/kube-prometheus-stack/templates/prometheus-operator/admission-webhooks/job-patch/psp.yaml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 {{- if and .Values.prometheusOperator.enabled .Values.prometheusOperator.admissionWebhooks.enabled .Values.prometheusOperator.admissionWebhooks.patch.enabled .Values.global.rbac.create .Values.global.rbac.pspEnabled (not .Values.prometheusOperator.admissionWebhooks.certManager.enabled) }}
+{{- if .Capabilities.APIVersions.Has "policy/v1beta1/PodSecurityPolicy" }}
```
Comment (monotek) [inline charts/kube-prometheus-stack/templates/prometheus-operator/admission-webhooks/job-patch/psp.yaml:2]: Please add to if clause above

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by QuentinBisson** [comment]
@monotek it's fixed

**Review by monotek — APPROVED** [review]

**Pull request merged**